test/unit/test-custom-element.js

```diff
@@ -1493,6 +1493,106 @@ describes.realWin('CustomElement', {amp: true}, (env) => {
           expect(testElementResumeCallback).to.have.not.been.called;
         });
       });
+
+      describe('ensureLoaded', () => {
+        it('should build and load', async () => {
+          const element = new ElementClass();
+          element.setAttribute('layout', 'fixed');
+          element.setAttribute('width', '10');
+          element.setAttribute('height', '10');
+          container.appendChild(element);
+          const resource = element.getResource_();
+
+          expect(element.isBuilt()).to.be.false;
+
+          const parentPriority = 1;
+          resourcesMock
+            .expects('scheduleLayoutOrPreload')
+            .withExactArgs(
+              resource,
+              /* layout */ true,
+              parentPriority,
+              /* forceOutsideViewport */ true
+            )
+            .once();
+
+          const promise = element.ensureLoaded(parentPriority);
+          await element.buildInternal();
+          await element.layoutCallback();
+
+          await promise;
+          expect(element.isBuilt()).to.be.true;
+        });
+
+        it('should load pre-built element', async () => {
+          const element = new ElementClass();
+          element.setAttribute('layout', 'fixed');
+          element.setAttribute('width', '10');
+          element.setAttribute('height', '10');
+          container.appendChild(element);
+          const resource = element.getResource_();
+
+          await element.buildInternal();
+          expect(element.isBuilt()).to.be.true;
+
+          const parentPriority = 1;
+          resourcesMock
+            .expects('scheduleLayoutOrPreload')
+            .withExactArgs(
+              resource,
+              /* layout */ true,
+              parentPriority,
+              /* forceOutsideViewport */ true
+            )
+            .once();
+
+          const promise = element.ensureLoaded(parentPriority);
+          await element.layoutCallback();
+
+          await promise;
+        });
+
+        it('should do nothing for already-loaded element', async () => {
+          const element = new ElementClass();
+          element.setAttribute('layout', 'fixed');
+          element.setAttribute('width', '10');
+          element.setAttribute('height', '10');
+          container.appendChild(element);
+
+          await element.buildInternal();
+          await element.layoutCallback();
+
+          resourcesMock.expects('scheduleLayoutOrPreload').never();
+
+          await element.ensureLoaded();
+        });
+
+        it('should do nothing for a non-displayed element', async () => {
```

Comment:
FMI: all users of Owners system are expected to call "requireLayout` every time the elements may have moved around s.t. a new one Resource is displayed?

Reply:
No. `requireLayout` is only called by the likes of animations to avoid animating empty boxes.